### PR TITLE
[FIX] point_of_sale: correctly add Global Discount with fiscal position

### PR DESCRIPTION
**Pull request by pedrambiria**
Previously, it was not possible to apply the Global Discount in cases where there was a fiscal position with tax included in the price. This was because of the `get_taxes` function, which do not return taxes after fiscal position.

opw-3267243

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_discount/static/src/js/DiscountButton.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ odoo.define('pos_discount.DiscountButton', function(require) {
                 });
                 return;
             }
+            // Remove discount taxes
+            product.taxes_id = [];
```
Comment (caburj) [inline addons/pos_discount/static/src/js/DiscountButton.js:39]: Let's not do this. Let the user put tax on their discount product.

**Comment by caburj** [comment]
I'm generally okay with this fix.

File: addons/pos_discount/static/src/js/DiscountButton.js (hunk 1 above)
Comment (pedrambiria) [reply addons/pos_discount/static/src/js/DiscountButton.js:39]: @caburj Thanks for your review. 
It causes some issues if it contains a tax. Could you please explain me why they need to choose a tax for the discount product?

Comment (caburj) [reply addons/pos_discount/static/src/js/DiscountButton.js:39]: I received comments just few days after I returned to pos team, that there are pos users who purposely put tax to their discount products. So unless we're very sure about that override (remove directly the tax specified by the user), then we should not do it.

Comment (pedrambiria) [reply addons/pos_discount/static/src/js/DiscountButton.js]: @caburj I removed it.

**Review by caburj — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**